snuba/query/processors/uniq_killswitch.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import annotations
+
+from snuba import state
+from snuba.clickhouse.processors import QueryProcessor
+from snuba.clickhouse.query import Query
+from snuba.query.expressions import Expression, FunctionCall, Literal
+from snuba.request.request_settings import RequestSettings
+
+
+class UniqKillswitchProcessor(QueryProcessor):
+    """
+    This processor removes uniq() expressions from the query and replaces them with zero.
+    The reason for doing this is because uniq() queries are suspected to be causing ClickHouse to segfault.
+    Since it returns incorrect results, it is only to be used in case of emergency.
+
+    Example usage:
+    - Enable for all queries to the following tables
+        - uniq_killswitch_tables=errors_dist,errors_ro_dist
+    - Enable for referrer
+        - uniq_killswitch_referrers=referrer1,referrer2,referrer3
+
+    `uniq_killswitch_tables` and `uniq_killswitch_referrers` are independent
+    and uniq will be removed from all tables and referrers specified.
+    """
+
+    def process_query(self, query: Query, request_settings: RequestSettings) -> None:
+        table_config = state.get_config("uniq_killswitch_tables")
+        referrer_config = state.get_config("uniq_killswitch_referrers")
```

Comment:
Can we have a column config and a project config as well? I think it is ok to do in a followup.
It is likely that only one column is the root cause. So most likely uniq(group_id) is fine while uniq(user_id) isn't.

Reply:
ok, I will do that in a separate PR because that increases the complexity